test/base_test.rb

```diff
@@ -33,13 +34,13 @@ def get_component_generation_command(component)
     btx_component_upstream_model: '-u',
     btx_component_usr_header_file: '-i'
   }
-  str_ = component.filter_map { |k, v| "#{args[k]} #{ [v].flatten.join(',') }" if args.key?(k) }.join(' ')
+  str_ = component.filter_map { |k, v| "#{args[k]} #{[v].flatten.join(',')}" if args.key?(k) }.join(' ')
   "ruby -I./lib ./bin/metababel #{str_}"
 end
 
 def get_component_compilation_command(component)
   command = <<~TEXT
-    ${CC:-cc} -o #{component[:btx_component_path]}/#{component[:btx_pluggin_name]}_#{component[:btx_component_name]}.so
+    ${CC:-cc} -o #{component[:btx_component_path]}/#{component[:btx_component_type]}_#{component[:btx_plugin_name]}_#{component[:btx_component_name]}.so
```

Comment:
If the user specify btx_plugin_name=plugin it will render something like `FILTER_plugin_component` if the user don't it will be something like `FILTER_plugin_component`, did you expect this ?

Reply:
Yeah doesn't matter just needs to be unique.

Reply:
But let use the compact, join to be clean :)

Reply:
Fixed by 5f02154